scripts/partition_manager.py

```diff
@@ -512,33 +504,90 @@ def get_dynamic_area_start_and_size(static_config, flash_size):
     return start, end - start
 
 
-def get_pm_config(input_config, flash_start, flash_size, static_config):
-    to_resolve = dict()
-    start = flash_start
+def get_region_config(pm_config, region_config, static_conf=None):
+    start = region_config['base_address']
+    size = region_config['size']
+    placement_strategy = region_config['placement_strategy']
+    region_name = region_config['name']
+    device = region_config['device']
+
+    if region_config['placement_strategy'] in [END_TO_START, START_TO_END]:
+        solve_simple_region(pm_config, start, size, placement_strategy, region_name, device, static_conf)
+    else:
+        solve_complex_region(pm_config, start, size, placement_strategy, region_name, device, static_conf)
+
+
+def solve_simple_region(pm_config, start, size, placement_strategy, region_name, device, static_conf):
+    reserved = 0
+    if static_conf:
+        verify_static_conf(size, start, placement_strategy, static_conf)
+        reserved = sum([config['size'] for name, config in static_conf.items()
+                        if 'region' in config.keys() and config['region'] == region_name and name != 'app'])
+
+    if placement_strategy == END_TO_START:
+        address = start + size - reserved
+    else:
+        address = start + reserved
+    for partition_name in pm_config:
+        if placement_strategy == END_TO_START:
+            address -= pm_config[partition_name]['size']
+
+        pm_config[partition_name]['address'] = address
 
-    load_reqs(to_resolve, input_config)
-    free_size = flash_size
+        if placement_strategy == START_TO_END:
+            address += pm_config[partition_name]['size']
 
-    if static_config:
-        start, free_size = get_dynamic_area_start_and_size(static_config, free_size)
-        to_resolve = {name: config for name, config in to_resolve.items()
-                      if name not in static_config.keys() or name == 'app'}
+        if device:
+            pm_config[partition_name]['device'] = device
+
+    # Generate the region partition containing the non-reserved memory
+    pm_config[region_name] = dict()
+    pm_config[region_name]['region'] = region_name
+
+    if placement_strategy == END_TO_START:
+        pm_config[region_name]['address'] = start
+        pm_config[region_name]['size'] = address - start
+    else:
+        pm_config[region_name]['address'] = start
+        pm_config[region_name]['size'] = (start + size) - address
```

Comment:
This doesn't seem quite right :P, but I'm not quite sure what is meant by "non-reserved memory", should it be something like this?

```suggestion
    if placement_strategy == END_TO_START:
        pm_config[region_name]['address'] = address
        pm_config[region_name]['size'] = start + size - reserved - address
    else:
        pm_config[region_name]['address'] = start + reserved
        pm_config[region_name]['size'] = address - start - reserved
```

Reply:
The non-reserved memory is whatever is not set aside for any specified partition. So the logic is correct, as is verified by the unit test.

```
    # Verify that RAM configuration is correct
    td = {'b': {'size': 100, 'region': 'ram'}}
    test_region = {'name': 'ram',
                   'size': 1000,
                   'base_address': 2000,
                   'placement_strategy': END_TO_START,
                   'device': None}
    get_region_config(td, test_region)
    assert td['b']['size'] == 100
    assert td['ram']['address'] == 2000
    assert td['ram']['size'] == 900
```

Here you see that the 'ram' partition which contain the non-reserved memory is created from the start of the region to the first defined partition.

Reply:
I see, thanks. I think there's only a small issue with START_TO_END (see updated suggestion). I suggest adding tests for START_TO_END as well (I couldn't find any).

Reply:
BTW you should reword the comment since `reserved` is a separate concept in this function.

Reply:
Nice spot! I added the test and fixed the bug.